packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -109,12 +109,19 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
 
   @autobind
   public dismissSuggestions() {
-    // Select the first suggestion if one is available when user leaves.
     let selectItemFunction = () => {
-      if (this.suggestionStore.hasSelectedSuggestion() && this.state.suggestedDisplayValue) {
-        this.addItemByIndex(0);
+      if (this.props.onDismiss) {
+        this.props.onDismiss(
+          this.suggestionStore.currentSuggestion ? this.suggestionStore.currentSuggestion.item : undefined
+        );
+      } else {
```

Comment:
You are still removing the default behavior anytime someone passes in an onDismiss. Some people will probably use onDismiss for other things (analytics, or whatever) and still want the default behavior.

You can provide this by calling this.props.onDismiss if it exists, and then running the default block of code afterwards gated by `!ev.defaultPrevented && defaultStuff`.

Check out this example. https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx#L491

Reply:
@micahgodbolt How would the user of this component prevent the default behavior? The event comes from the protected `Callout` and would require overriding the `renderSuggestions` method. 

Wouldn't it be easier to introduce a the flag `preventDefault` without the event?

Reply:
Now I got it right!